File: OC/20190606/Configuration/UnitClass/Reachability/XXZNet.m
//
//  XXZNet.m
//  Author From Xavier Zachary
//
//  Created by Zachary on 14/4/10.
//  Copyright © 2015年 com.xxzing.future. All rights reserved.
//

#import "XXZNet.h"

@implementation XXZNet

//返回网络类型
+ (NetworkStatus)netType {
    //初始化网络
    Reachability *baiduReach = [Reachability reachabilityWithHostname:@"www.baidu.com"];
    [baiduReach startNotifier];
    
    //网络状态
    NetworkStatus status = baiduReach.currentReachabilityStatus;
    //NotReachable: //0 //无网络
    //ReachableViaWiFi: //2 //WIFI
    //ReachableViaWWAN: //1 //蜂窝
    
    return status;
}

@end

File: Swift/20190827/Configuration/Encapsulation/Network/Reachability/XXZNet.swift
//
//  XXZNet.swift
//  Author From Xavier Zachary
//
//  Created by Zachary on 2017/8/1.
//  Copyright © 2017年 Zachary. All rights reserved.
//

import UIKit

public let isNetwork = XXZNet.isNetwork()
public let networkStatus = XXZNet.returnNetworkStatus()

private class XXZNet: NSObject {
    private override init() {}
    
    //MARK: ------ Zachary - net ------
    static func returnNetworkStatus() -> Int { //网络类型
        let reach = Reachability.init(hostname: "www.baidu.com")
        let status = reach?.currentReachabilityStatus
        
        var networkStatus = -1
        
        if status == .reachableViaWiFi {
            networkStatus = 1
        }
        else if status == .reachableViaWWAN {
            networkStatus = 2
        }
        else if status == .notReachable {
            networkStatus = 0
        }
        
        return networkStatus
    }
    
    static func isNetwork() -> Bool { //有无网络
        let reach = Reachability.init(hostname: "www.baidu.com")
        let status = reach?.currentReachabilityStatus
        
        var isNetwork = true
        if status == .notReachable {
            isNetwork = false
        }
        
        return isNetwork
    }
}
